### PR TITLE
MDTZ-1512: Handle permission-related errors from Jira when handling backward integration with Figma

### PR DESCRIPTION
**Pull request by akostevich-atlassian**
### Changes

- **fix:** Resolves the issue with the "_Backfill Designs_", "_Associate Design_" and "_Disassociate Design_" use cases failing when the app is handling backward integration with Figma but does not have permission to read/modify Jira Issues. Updates the use cases to handle permission-related errors from Jira and make _the best effort_ to support backward integration.

image

**NOTE:** Strictly speaking, the app always handled the backward integration as "the best effort" as Figma API has its own restrictions (e.g., no more than 10 Dev Resources per design). Therefore, the PR relaxes the requirements further to avoid failures in the valid business scenarios.

### CODE REVIEW

File: src/infrastructure/jira/jira-client/jira-client.test.ts [hunk 1]
```diff
@@ -26,14 +26,13 @@ jest.mock('./jwt-utils');
 
 describe('JiraClient', () => {
 	let connectInstallation: ConnectInstallation;
-	const createJwtTokenMock = jest.mocked(createJwtToken);
```
Comment (akostevich-atlassian) [inline src/infrastructure/jira/jira-client/jira-client.test.ts:29]: **note:** Just a minor refactoring.

File: src/infrastructure/jira/jira-design-issue-property-service.ts [hunk 2]
```diff
@@ -46,55 +46,71 @@ const ATTACHED_DESIGN_URL_V2_VALUE_SCHEMA: JSONSchemaTypeWithId<AttachedDesignUr
 	};
 
 export class JiraDesignIssuePropertyService {
-	/**
-	 * @throws {ForbiddenByJiraServiceError} The app does not have permission to edit the Issue.
-	 */
-	saveDesignUrlInIssueProperties = async (
+	trySaveDesignUrlInIssueProperties = async (
```
Comment (akostevich-atlassian) [inline src/infrastructure/jira/jira-design-issue-property-service.ts:49]: **note:** Converted `trySaveDesignUrlInIssueProperties` to `saveDesignUrlInIssueProperties` by analogue with `figmaService.tryCreateDevResourceForJiraIssue` to highlight that the method makes the best effort to perform the task.

File: src/usecases/associate-design-use-case.ts [hunk 3]
```diff
@@ -46,54 +46,42 @@ export const associateDesignUseCase = {
...
+
 			await Promise.all([
-				jiraService.submitDesign(
+				await associatedFigmaDesignRepository.upsert({
+					designId: figmaDesignId,
+					associatedWithAri: associateWith.ari,
+					connectInstallationId: connectInstallation.id,
+					inputUrl: designUrl.toString(),
+				}),
+				await figmaBackwardIntegrationService.tryNotifyFigmaOnAddedIssueDesignAssociation(
```
Comment (stannnous) [inline src/usecases/associate-design-use-case.ts:74]: Praise: nice refactor moving this responsibility out of the use case

**Review by stannnous — APPROVED** [review]

**Review by Roystbeef — APPROVED** [review]
amazing - thanks for doing this @akostevich-atlassian!

**Pull request merged**